dali/operators/generic/constant.h

```diff
@@ -80,9 +80,9 @@ class Constant : public Operator<Backend> {
 
   bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override {
     output_desc.resize(1);
+    auto curr_batch_size = ws.GetRequestedBatchSize(0);
     if (output_shape_.empty()) {
```

Comment:
```suggestion
    if (output_shape_.empty() || output_shape_ .size() < curr_batch_size ) {
```
Because if you the rest iteration has batch size 1 and the next one 2, you will end up with `output_shape_ ` os size 1 once and for all.